// bloxlink-api-sdk/dist/BloxlinkGuildClient.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
const fetch = (_a = globalThis.fetch) !== null && _a !== void 0 ? _a : require("node-fetch");
class BloxlinkGuildClient {
    constructor(options) {
        this.options = options;
    }
    discordToRoblox(discordUserId, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const apiOptions = options !== null && options !== void 0 ? options : this.options;
            const data = yield fetch(`https://api.blox.link/v4/public/guilds/${apiOptions.guildId}/discord-to-roblox/${discordUserId}`, {
                headers: { Authorization: apiOptions.apiKey },
            }).then(res => res.json());
            return data;
        });
    }
    robloxToDiscord(robloxUserId, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const apiOptions = options !== null && options !== void 0 ? options : this.options;
            const data = yield fetch(`https://api.blox.link/v4/public/guilds/${apiOptions.guildId}/roblox-to-discord/${robloxUserId}`, {
                headers: { Authorization: apiOptions.apiKey },
            }).then(res => res.json());
            return data;
        });
    }
    updateUser(discordUserId, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const apiOptions = options !== null && options !== void 0 ? options : this.options;
            const data = yield fetch(`https://api.blox.link/v4/public/guilds/${apiOptions.guildId}/update-user/${discordUserId}`, {
                headers: { Authorization: apiOptions.apiKey },
                method: "POST",
            }).then(res => res.json());
            return data;
        });
    }
    set setAPIOptions(options) {
        this.options = options;
    }
    get apiOptions() {
        return this.options;
    }
}
exports.default = BloxlinkGuildClient;
